onadata/apps/viewer/tasks.py

```diff
@@ -96,18 +96,25 @@ def _create_export(xform, export_type, options):
         Export.EXTERNAL_EXPORT: create_external_export
     }
 
-    # start async export
-    if export_type in export_types:
-        try:
-            result = export_types[export_type].apply_async((), kwargs=options)
-        except OperationalError as e:
-            export.internal_status = Export.FAILED
-            export.error_message = "Error connecting to broker."
-            export.save()
-            report_exception(export.error_message, e, sys.exc_info())
-            raise Export.ExportConnectionError
+    if export.xform.num_of_submissions > 0:
+        # start async export
+        if export_type in export_types:
+            try:
+                result = export_types[export_type].apply_async(
+                                        (), kwargs=options)
+            except OperationalError as e:
+                export.internal_status = Export.FAILED
+                export.error_message = "Error connecting to broker."
+                export.save()
+                report_exception(export.error_message, e, sys.exc_info())
+                raise Export.ExportConnectionError
+        else:
+            raise ExportTypeError
     else:
-        raise ExportTypeError
+        export.error_message = "No records found for this export"
+        export.internal_status = Export.FAILED
+        export.save()
```

Comment:
I think we should out-right block the creation of an Export object if the target XForm has no Submissions; The request export task should not be queued

Reply:
@DavisRayM When the media file is not generated it means the user may not even be able to download the form for data collection. Generating a file with only headers should be a valid export imo.

Reply:
Included changes that generate csv export file with only the headers.
Removed the checks below as well.
Thanks for this!